src/dimension_slice.h

```diff
@@ -1,6 +1,9 @@
 #ifndef TIMESCALEDB_DIMENSION_SLICE_H
 #define TIMESCALEDB_DIMENSION_SLICE_H
 
+#define DIMENSION_SLICE_MAXVALUE ((int64)PG_INT32_MAX + 1)
```

Comment:
So, from what I understand, we can never hash to this max and min value, so why do we need them to be +1 and -1 if our CHECK constraints are already open-ended. So, if a slice has value `PG_INT32_MAX` we will have a check constraint `CHECK(value >= RANGE_START)`, where `RANGE_START` is value less than `PG_INT32_MAX`. Changing the check constraints seems to be enough to fix this problem without altering the ranges. Am I missing something?

Reply:
So `DIMENSION_SLICE_MAXVALUE` has to be > than the max value given by hash (which is `PG_INT32_MAX` I believe) because semantically range_end is > (not >= than the range).

Reply:
However `DIMENSION_SLICE_MINVALUE` can be 0. I struggled whether to make it -1 or 0. Don't know whats better.